oap-server/server-query-plugin/query-graphql-plugin/src/main/java/org/apache/skywalking/oap/query/graphql/resolver/MetricQuery.java

```diff
@@ -59,17 +63,24 @@ public IntValues getValues(final BatchMetricConditions metrics, final Duration d
             kv.setValue(query.readMetricsValue(condition, duration));
             values.addKVInt(kv);
         } else {
-            for (final String id : metrics.getIds()) {
-                KVInt kv = new KVInt();
-                kv.setId(id);
+            List<CompletableFuture<KVInt>> futureList = metrics.getIds().stream().map(id -> {
 
                 MetricsCondition condition = new MetricsCondition();
                 condition.setName(metrics.getName());
                 condition.setEntity(new MockEntity(id));
-
-                kv.setValue(query.readMetricsValue(condition, duration));
-                values.addKVInt(kv);
-            }
+                CompletableFuture<KVInt> future = CompletableFuture.supplyAsync(() -> {
+                    try {
+                        KVInt kv = new KVInt();
+                        kv.setId(id);
+                        kv.setValue(query.readMetricsValue(condition, duration));
+                        return kv;
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+                return future;
+            }).collect(Collectors.toList());
+            futureList.stream().map(CompletableFuture::join).forEach(v -> values.addKVInt(v));
```

Comment:
Actually `CompletableFuture.supplyAsync` can be replaced with `parallelStream` with just a very trivial difference that it doesn't matter in the case here, but is simpler, what do you think?

```java
            metrics.getIds()
                   .parallelStream()
                   .map(id -> {
                        MetricsCondition condition = new MetricsCondition();
                        condition.setName(metrics.getName());
                        condition.setEntity(new MockEntity(id));
                        try {
                            KVInt kv = new KVInt();
                            kv.setId(id);
                            kv.setValue(query.readMetricsValue(condition, duration));
                            return kv;
                        } catch (IOException e) {
                            throw new RuntimeException(e);
                        }
                   })
                   .forEach(v -> values.addKVInt(v));

```

Reply:
@chenhaipeng What do you think about this?

Reply:
yes! I think so too

Reply:
OK, I would wait for your update.

Thanks for bringing this to the community's attention. We missed this when building `v1->v2` bridge.